src/inmanta/parser/plyInmantaParser.py

```diff
@@ -692,6 +692,33 @@ def p_boolean_expression_is_defined_short(p: YaccProduction) -> None:
     attach_lnr(p)
 
 
+def p_boolean_expression_is_defined_map_lookup_0(p: YaccProduction) -> None:
+    """boolean_expression : is_defined_expr IS DEFINED"""
+    p[0] = p[1]
+
+
+def p_is_defined_expr_is_defined_map_lookup_1(p: YaccProduction) -> None:
+    """is_defined_expr : var_ref '[' operand ']'
+    | attr_ref '[' operand ']'
+    | map_lookup '[' operand ']'"""
+
+    location = Location(file, p.lineno(2))
+    lexpos = p.lexpos(2)
+
+    def attach_lnr_to_statement(inp: ExpressionStatement) -> ExpressionStatement:
+        "set location"
+        inp.location = location
+        inp.lexpos = lexpos
+        return inp
+
+    key_in_dict = attach_lnr_to_statement(In(p[3], p[1]))
```

Comment:
Let's add a comment here.
```suggestion
    # syntactic sugar for `is defined` on dicts
    key_in_dict = attach_lnr_to_statement(In(p[3], p[1]))
```